activestorage/lib/active_storage/validations/attachment_content_type_validator.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+module ActiveStorage
+  module Validations
+    class AttachmentContentTypeValidator < BaseValidator
+      AVAILABLE_CHECKS = %i[in not]
+
+      def self.helper_method_name
+        :validates_attachment_content_type
+      end
+
+      def initialize(options = {})
+        # <tt>ActiveModel::Validations#validates</tt> allows for shortcut
+        # options. It automatically stores ranges and arrays in
+        # <tt>options[:in]</tt> and everything else in <tt>options[:with]</tt>.
+        # Since content type validation allows specifying a single content
+        # type as a string it needs to be manually captured here.
+        options[:in] ||= options[:with] if options[:with].is_a?(String)
+
+        super
+      end
+
+      def validate_each(record, attribute, value)
+        @record = record
+        @name = attribute
+
+        error_key_for_check_name = { in: :inclusion, not: :exclusion }
+
+        each_blob do |blob|
+          each_check do |check_name, check_value|
+            next if passes_check?(blob, check_name, check_value)
+
+            error_key = error_key_for_check_name[check_name.to_sym]
+            record.errors.add(@name, error_key, error_options)
+          end
+        end
+      end
+
+      def check_validity!
+        if options_blank?
+          raise(
+            ArgumentError,
+            "You must pass either :in or :not to the validator"
+          )
+        end
+
+        if options_redundant?
+          raise(ArgumentError, "Can not pass both :in and :not")
```

Comment:
(Will also need to update the test case that uses this string)
```suggestion
          raise(ArgumentError, "Cannot pass both :in and :not")
```